src/convertToObject.js

```diff
@@ -5,8 +5,25 @@
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
+function convertToObject(str) {
+  const resultStyles = {};
+  const listStyle = str.split(';');
+
+  for (const listStyleElement of listStyle) {
+    const styleItem = listStyleElement.trim();
+
+    if (styleItem) {
+      const [property, ...itemParts] = styleItem.split(':');
+
+      if (property && itemParts.length > 0) {
+        const item = itemParts.join(':').trim();
+
+        resultStyles[property.trim()] = item;
+      }
+    }
```

Comment:
The use of a for...of loop here goes against the checklist item 5, which advises to avoid using loops and use iteration methods instead. Consider using array methods like 'filter', 'map', and 'reduce' to transform the string into the desired object.